packages/governance/src/contractGovernance/paramManager.js

```diff
@@ -1,415 +1,618 @@
+        const { state } = this;
+        return {
+          type: state.type,
+          value: state.current,
+        };
+      },
+      getValue() {
+        return this.state.current;
+      },
+      getVisibleValue(proposed) {
+        const {
+          state: { name, pattern },
+        } = this;
+        mustMatch(
+          proposed,
+          pattern,
+          `Proposed value ${proposed} for ${name} must match ${pattern}`,
+        );
+        return proposed;
+      },
+      prepareToUpdate(proposed) {
+        return proposed;
+      },
+      update(proposed) {
+        const { state } = this;
+        mustMatch(
+          proposed,
+          state.pattern,
+          `${state.name} must match ${state.pattern}, was ${proposed}`,
+        );
+        state.current = proposed;
+        return harden({ [state.name]: proposed });
+      },
+      getGetterElement() {
+        const { state } = this;
+        // names are keywords so they will necessarily be TitleCase
+        const name = `get${state.name}`;
+        const getter = () => state.current;
+        return [name, getter];
+      },
+      getFarGetterElement() {
+        const { state } = this;
+        // names are keywords so they will necessarily be TitleCase
+        const name = `get${state.name}`;
+        const getter = ToFarFunction(name, () => state.current);
+        const guard = M.call().returns(state.pattern);
+        return [name, getter, guard];
+      },
+    },
+    {
+      stateShape: {
+        name: KeywordShape,
+        current: M.any(),
+        // XXX how to say it must be one of ParamTypes?
+        type: M.string(),
+        pattern: M.pattern(),
+      },
+    },
+  );
+
+  const assertInvitation = async i => {
+    if (!zoe) {
+      throw Fail`zoe must be provided for governed Invitations`;
+    }
+    const { instance, installation } = await E(zoe).getInvitationDetails(i);
+    // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error -- the build config doesn't expect an error here
+    // @ts-ignore typedefs say they're guaranteed truthy but just to be safe
+    assert(instance && installation, 'must be an invitation');
+  };
+
+  const InvitationParamHolderI = M.interface('InvitationParamHolder', {
+    ...ParamHolderCommon,
+    getValue: M.call().returns(InvitationShape),
+  });
+
+  const makeInvitationParamHolder = prepareExoClass(
+    baggage,
+    'InvitationParamHolder',
+    InvitationParamHolderI,
+    (name, current, amount) => {
+      return { name, current, amount };
+    },
+    {
+      makeDescription() {
+        const { state } = this;
+        return {
+          type: ParamTypes.INVITATION,
+          value: state.amount,
+        };
+      },
+      async prepareToUpdate(invite) {
+        const { state } = this;
+        const [preparedAmount] = await Promise.all([
+          E(E(zoe).getInvitationIssuer()).getAmountOf(invite),
+        ]);
+        assertInvitation(invite);
```

Comment:
that's an async function so much be awaited to throw upon failure. I think this is a bug in master too.